torch/_dynamo/variables/builder.py

```diff
@@ -432,29 +428,30 @@ class Autotuner:
             else:
                 guards = self.make_guards(GuardBuilder.DICT_KEYS)
 
-            # store key variables in global location for reconstruction
-            for key in value.keys():
-                if self.tensor_can_be_dict_key(key):
-                    self.tx.store_global_weakref(global_key_name(key), key)
+            idx = 0
 
-            def index_source(key):
-                if self.tensor_can_be_dict_key(key):
-                    return GlobalWeakRefSource(global_key_name(key))
+            def build_key_value(k, v):
+                nonlocal idx
+                if ConstantVariable.is_literal(k):
+                    key = ConstantVariable.create(k).add_guards(guards)
+                    source_key = k
                 else:
-                    return key
+                    source_key = ConstDictKeySource(self.get_source(), idx)
```

Comment:
I believe this relies on the order of the dictionary keys, however I don't think we are guarding on dict key order.

Please add a test where you call a function with two dictionaries with the *same* keys but *different* orders.